### PR TITLE
Updates references to security demo configuration

### PR DESCRIPTION
**Pull request by DarshitChanpura**
### Description
_Describe what this change achieves._

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by DarshitChanpura** [comment]
@Xtansia @VachaShah I would need your help to ensure that this instance where the demo script is being called: 
https://github.com/opensearch-project/opensearch-net/blob/f1df48dad8044121dd02f5ab133b349162026f81/abstractions/src/OpenSearch.OpenSearch.Ephemeral/Tasks/InstallationTasks/InitialConfiguration.cs#L52

**Comment by DarshitChanpura** [comment]
If I'm interpreting the github actions correctly, the integration tests do not run with security correct?

**Comment by derek-ho** [comment]
https://github.com/opensearch-project/security/issues/3624

**Comment by derek-ho** [comment]
We can merge this once it's out of draft

**Comment by DarshitChanpura** [comment]
@Xtansia would you help bring this across finish line?

**Comment by dblock** [comment]
@DarshitChanpura what remains to be done? Mark it as ready if you're ready.

**Comment by Xtansia** [comment]
@DarshitChanpura @dblock I have been working on this, but I'm currently still on leave until the 8th atm. This shouldn't block the 2.12 release as we don't automatically pick up the new version here and the client is not in sync with core versioning/releasing anyways

**Comment by Xtansia** [comment]
@DarshitChanpura @dblock This is now ready for review and passing tests against unreleased 2.x & main.

**Review by DarshitChanpura — COMMENTED** [review]
Thank you @Xtansia for taking this over.

File: abstractions/src/OpenSearch.OpenSearch.Ephemeral/Tasks/InstallationTasks/InitialConfiguration.cs [hunk 1]
```diff
@@ -38,30 +40,39 @@ public class InitialConfiguration : ClusterComposeTask
...
 
+			if (File.Exists(installConfigFile) && File.ReadLines(installConfigFile).Any(l => l.Contains("plugins.security"))) return;
+
+			var env = new Dictionary<string, string>();
+			var args = new List<string> { securityInstallDemoConfig, "-y", "-i" };
+
+			if (isNewDemoScript)
+			{
+				env.Add("OPENSEARCH_INITIAL_ADMIN_PASSWORD", "admin");
+				args.Add("-t");
```
[the remark this answers is not in the document]
Comment (Xtansia) [reply abstractions/src/OpenSearch.OpenSearch.Ephemeral/Tasks/InstallationTasks/InitialConfiguration.cs:66]: It's possible but requires pushing knowledge of the password and 2.12 version break further up the hierarchy. It also still wouldn't mean getting rid of `admin` it'd just mean conditionally setting it to a stronger password.

It's something we could revisit in a follow up PR if we do want the flexibility.

File: .github/workflows/integration.yml [hunk 2]
```diff
@@ -17,17 +17,14 @@ jobs:
       fail-fast: false
       matrix:
         version:
-          - 2.9.0
+          - 2.11.1
+          - 2.10.0
```
[the remark this answers is not in the document]
Comment (Xtansia) [reply .github/workflows/integration.yml:21]: Purely just for somewhat reducing the number of jobs

File: .github/actions/start-opensearch/action.yml [hunk 3]
```diff
@@ -34,7 +34,12 @@ runs:
 
         if [[ -d "$OPENSEARCH_HOME/plugins/opensearch-security" ]]; then
           if [[ "$SECURED" == "true" ]]; then
-            bash $OPENSEARCH_HOME/plugins/opensearch-security/tools/install_demo_configuration.sh -y -i -s
+            DEMO_CONFIG_SCRIPT="$OPENSEARCH_HOME/plugins/opensearch-security/tools/install_demo_configuration.sh"
+            if grep --quiet -i org.opensearch.security.tools.democonfig.Installer "$DEMO_CONFIG_SCRIPT"; then
```
[the remark this answers is not in the document]
Comment (Xtansia) [reply .github/actions/start-opensearch/action.yml:38]: I can change the check. We still have to support versions below 2.12, so we can't get rid of `admin` all together. So if we pass in a secure password, we'd still need the check as we still need to be able to differentiate whether the password will be the secure one or the `admin` one.

File: tests/Tests/Search/SearchTemplate/RenderSearchTemplate/RenderSearchTemplateApiTests.cs [hunk 4]
```diff
@@ -40,6 +40,7 @@
 
 namespace Tests.Search.SearchTemplate.RenderSearchTemplate
 {
+	[SkipVersion("2.10.*,2.11.*", "Broken by security plugin https://github.com/opensearch-project/security/issues/3672")]
```
[the remark this answers is not in the document]
Comment (DarshitChanpura) [reply tests/Tests/Search/SearchTemplate/RenderSearchTemplate/RenderSearchTemplateApiTests.cs:43]: okay.

**Review by DarshitChanpura — COMMENTED** [review]
LGTM. Thank you once again @Xtansia.

**Comment by Xtansia** [comment]
@dblock Could we please get your review on this?

**Comment by dblock** [comment]
> @dblock Could we please get your review on this?

This isn't needed until 2.12 and should not be passing until then, shouldn't it?

**Comment by Xtansia** [comment]
@dblock We do actually already hit the code paths in this PR because we test against 2.x and main with the security plugin installed. It's just that the tests against unreleased OpenSearch had been disabled due to breakage, which is now fixed and re-enabled in this PR. So all that's left for follow up is adding `2.12` to the released versions matrix when it's released

**Review by dblock — APPROVED** [review]

**Pull request merged**